src/game/utils/direction.ts

```diff
@@ -0,0 +1,43 @@
+import Position from "../world/position";
+export type Direction = "UP" | "DOWN" | "LEFT" | "RIGHT" | "NORTH" | "SOUTH" | "WEST" | "EAST";
```

Comment:
You use lowercase everywhere else, I would suggest that you make these lowercase too.
Also, why not use an enum? This is literally what they are for

Reply:
I'm not sure if I use lowercase everywhere else. I tend to use uppercase for constants, and I would still use these for enums. I'm not really too familiar with TS implementation of enums and it seems that what I want is either:
```
export enum Direction {
  NORTH = "north",
  SOUTH = "south",
  EAST = "east",
  WEST = "west",
}
```
or alternatively not use the strings and just let it give those types numbers:
```
export enum Direction {
  NORTH, SOUTH, EAST, WEST,
}
```
Do you know a reason why one would be preferable? Also what's the benefit of using an enum over a union type here?

Reply:
I would leave it as numbers.

Reply:
Why? If using strings, if you console.log(Direction.North) you get the string instead of a seemingly arbitrary number!

Reply:
Yeah, actually strings are probably better with ts enums

Reply:
Yeah that's fair, I think having it as it is currently is ok. I don't think I'm using the string for anything, it's just better in case it crops up in an error message or I need to log it; I won't use it for anything else.

Reply:
> it's just better in case it crops up in an error message or I need to log it; I won't use it for anything else.

Why not?

Reply:
Turns out I was using this in some JS, which meant that I couldn't import the enum and use it and I didn't want to use numbers because that made it too obscure. Having the strings made that code readable: https://github.com/Trymunx/dragon-slayer/pull/13/commits/a31aef95978a137c346924c847928d8541082b68#diff-8d8dba2ce00c2c440d42d3db30240dd0

Reply:
Do they not just import to js as strings?

Reply:
I didn't try, I just used the string representation instead. I'll be converting the file to TS at some point soon anyway so I'm not worried.